src/app/shared/services/authentication.service.ts

```diff
@@ -1,34 +1,54 @@
 import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
 
-import { User } from '../interfaces/user.interface.';
+import { BehaviorSubject, Observable, Subject, last } from 'rxjs';
+
+import { HttpUser } from '../interfaces/http-user.interface';
 
 @Injectable({
   providedIn: 'root'
 })
 export class AuthenticationService {
 
-  loggedUser!: User | null;
-
-  login(user: User): void {
-    this.loggedUser = {...user};
-    localStorage.setItem("token", JSON.stringify(this.loggedUser));
-    window.location.reload();
+  loggedUser!: HttpUser | null;
+  private apiUrl = 'http://localhost:3004';
+  userSubject = new Subject<HttpUser>();
+  isAuthSubject = new BehaviorSubject<boolean>(false);
+  isAuthenticated$: Observable<boolean> = this.isAuthSubject.asObservable();
+
+  constructor(private http: HttpClient) {}
+
+  login(login: string, password: string){
+    this.http
+      .post(
+        `${this.apiUrl}/auth/login`,
+        {login, password},
+      )
+      .subscribe( response => {
```

Comment:
You never should subscribe in the service - it is considered the bad practice
All actions which are needed to be provided here - must be done the help of **RxJS**

Reply:
ok, moved the logic to the component